docs/source/containers.rst

```diff
@@ -1,34 +1,303 @@
+^^^^^^^^^^^^^^^^^^^
+
+A container's primary role is containing the information necessary to load the
+Python representation of an object.
+
+Internally, happi keeps track of all containers by way of its registry, the
+:class:`~happi.containers.HappiRegistry`.
+
+This information is stored as a ``device_class``, ``args`` and ``kwargs``. The
+former stores a string that indicates the Python class of the item, the other
+two indicate the information that is needed to instantiate it. With this
+information both :func:`.from_container` and :func:`.load_device` will handle
+importing modules and instantiating your object.
+
+.. note::
+
+    happi will attach the original metadata with a fixed attribute name ``.md``
+    to your object.  You can use this to keep track of the container metadata
+    used to instantiate your object.
+
+    This can be disabled by setting ``attach_md`` to ``False`` in
+    :func:`.from_container`.
+
+Often information contained in the ``args`` or ``kwargs`` will be duplicated in
+other parts of the container. For instance most ``ophyd`` objects will want a
+``name`` and ``prefix`` on initialization. Instead of repeating that
+information you can just use a template and have the information automatically
+populated for you by the container itself. For instance, in the aforementioned
+example ``container.args = ["{{name}}"]`` would substitute the attribute
+listed as ``container.name`` in as an argument. If the template contains the
+substituted attribute alone the type will also be converted.
+
+.. ipython:: python
+
+    from happi import from_container
+
+    container = MyItem(name="my_item", model_no="QABC1234",
+                       device_class='ophyd.sim.SynSignal',
+                       kwargs={'name': '{{name}}'})
+    obj = from_container(container, attach_md=True)
+    obj
+
+
+Integrating with your package
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Happi provides some containers for your convenience, but intentionally does
+not support all use cases or control systems in a monolithic fashion.
+
+The suggested method for supporting your new package would be to make your
+package have a dependency on happi, of course, and subclass `.HappiItem`
+to make a new container in your own package.
+
+Then, add an
+`entry point <https://packaging.python.org/specifications/entry-points/>`_
+specified by the **happi.containers** keyword to your package ``setup.py``.
+An example entry points can be found `here
+<https://github.com/pcdshub/pcdsdevices/blob/master/setup.py>`_.
+
+:class:`~happi.containers.HappiRegistry` takes care of loading the entry points
+and making them available throughout the library.
+
+
+.. _convention_label:
+
+Built-in Container Conventions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order for the database to be as easy to parse as possible we need to
+establish conventions on how information is entered. Please read through this
+before entering any information into the database.
+
+
+HappiItem Entries
```

Comment:
_maybe_ backtick the class name... _maybe_...
It's more helpful but it does look kinda ugly with the box around it in the heading